addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1489,7 +1489,7 @@ var SnippetsMenu = Widget.extend({
         // on text changes
         const alreadySelectedElements = new Set();
         this.$document.on('click.snippets_menu', '.o_default_snippet_text', function (ev) {
-            const el = ev.currentTarget;
+            const el = ev.target;
```

Comment:
https://github.com/odoo/odoo/pull/145925#discussion_r1458830385 continued

@Zynton The runbot failed due to rebasing. A recent PR https://github.com/odoo/odoo/pull/144121 [commit](https://github.com/odoo/odoo/pull/144121/commits/c6ce2e7a70c107f1622b71c870b02ff721c4da82) changed the behaviour of clicks on `o_default_snippet_text` to select the complete node when clicked. However, It uses `ev.currentTarget` as element. But for our mega menu case with snippet like
```html
<a class="o_default_snippet_text">
    <div>
        <i class=fa-xxx>​</i>
        <div>
            <h4>Text</h4>
            <font>Text</font>
        </div>
    </div>
</a>
```
When clicking on `h4` element, `ev.currentTarget` returns the `a` tag, as the property doesn't change during event propogation. regardless of where the event is originated. It would always refer to the element to which the event handler is attached, so in this case the `el` variable is set to the `a` tag rather than `h4` tag, causing the entire `a` tag to be selected when clicking on `h4`. So I have changed the el to `ev.target` instead, which represents the actual element on which the event occured. Would you agree with this change?? cc. @qsm-odoo

Reply:
Hello :wave: 

- I indeed changed `.target` to `.currentTarget` intentionally: "on click on a node marked as default text, select its content" being more accurate than "on click inside a node marked as default, select the inner node that was clicked". Especially since nothing guarantees that that "inner clicked" node is actually something the user actually sees: it could be a simple span without style for all we know :shrug: I probably should have mentioned it in the commit message. In any case, changing it back to `.target` does not seem to be the way to go (at least I don't understand your need).

- With my commit, I actually introduced new bugs that I still have to solve (unrelated to currentTarget/target) but the `o_default_snippet_text` is not removed properly anymore when it should after typing inside nodes, I still have to check that. That bug was reported by RDE here: https://github.com/odoo/odoo/pull/144121#discussion_r1438344278 but then I could not reproduce it, then he could not reproduce it anymore either... then after full forward-port, I found out there actually is the bug he described :shrug: Don't know how or why yet.

Reply:
I'm in agreement with @qsm-odoo here. What I'm wondering is (1) why it's not satisfactory to select the whole `<a>` as that seems like precisely the feature's goal; and (2) why that breaks your test.

Reply:
I intended to apply bold to `h4` in the tour. However, clicking on the `h4`, the selection moved to `a` tag, causing my tour to fail, Nevertheless, I have adapted the tour, it should be good now.